src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -249,10 +249,10 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
                             CustomModifierUtils.CopyTypeCustomModifiers(overriddenPropertyType.Type, type.Type, this.ContainingAssembly),
                             overriddenPropertyType.CustomModifiers);
 
-                        // Although we only do this in error scenarios, it is undesirable to mutate the symbol by setting its type twice.
+                        // It is undesirable to mutate the symbol by setting its type twice.
```

Comment:
>// It is undesirable to mutate the symbol by setting its type twice. [](start = 24, length = 68)

There is nothing wrong with this code, we are in a constructor. The comment should be removed and the issue is closed.

Reply:
@AlekseyTs The fact that we're in a constructor is not a reason why this code is safe.

Reply:
> The fact that we're in a constructor is not sufficient to prove this code is safe.

It is safe given what the code is doing, where it is doing that, and what kind of changes it makes to the value second time. It is not doing random arbitrary things.
We are inside a constructor and the previously assigned value is used only for the purpose of finding overridden/implemented methods. Also, the only part of the type that we are adjusting are custom modifiers, those in an overriding/implementing property have no impact what so ever on language semantics or on the way overridden/implemented methods are found.

---
In reply to: [280921420](https://github.com/dotnet/roslyn/pull/35459#discussion_r280921420) [](ancestors = 280921420)